troposphere/haze/source/event_reactor.cpp

```diff
@@ -49,6 +49,7 @@ namespace haze {
     }
 
     Result EventReactor::WaitForImpl(s32 *out_arg_waiter, const Waiter *arg_waiters, s32 num_arg_waiters) {
+        HAZE_ASSERT(0 <= num_arg_waiters && num_arg_waiters <= svc::ArgumentHandleCountMax);
```

Comment:
Is 0 supported? Does libnx do svc::CancelSynchronization stuff? Waiting on zero handles with an infinite timeout is an infinite wait on HOS. This is probably fine.

Reply:
Good catch, I don't intend to support 0 here